hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/StorageDistributionEndpoint.java

```diff
@@ -0,0 +1,199 @@
+ */
+@Path("/storagedistribution")
+@Produces("application/json")
+public class StorageDistributionEndpoint {
+  private final ReconNodeManager nodeManager;
+  private final OMDBInsightEndpoint omdbInsightEndpoint;
+  private final NSSummaryEndpoint nsSummaryEndpoint;
+  private final StorageContainerLocationProtocol scmClient;
+
+  @Inject
+  public StorageDistributionEndpoint(OzoneStorageContainerManager reconSCM,
+                                     OMDBInsightEndpoint omDbInsightEndpoint,
+                                     NSSummaryEndpoint nsSummaryEndpoint,
+                                     StorageContainerLocationProtocol scmClient) {
+    this.nodeManager = (ReconNodeManager) reconSCM.getScmNodeManager();
+    this.omdbInsightEndpoint = omDbInsightEndpoint;
+    this.nsSummaryEndpoint = nsSummaryEndpoint;
+    this.scmClient = scmClient;
+  }
+
+  @GET
+  public Response getStorageDistribution() {
+    List<DatanodeStorageReport> nodeStorageReports = collectDatanodeReports();
+    GlobalStorageReport globalStorageReport = calculateGlobalStorageReport();
+    Map<String, Long> namespaceMetrics = calculateNamespaceMetrics();
+    StorageCapacityDistributionResponse response = buildStorageDistributionResponse(
+        nodeStorageReports, globalStorageReport, namespaceMetrics);
+    return Response.ok(response).build();
+  }
+
+  private GlobalStorageReport calculateGlobalStorageReport() {
+    SCMNodeStat stats = nodeManager.getStats();
+    return new GlobalStorageReport(
+        stats.getScmUsed().get(),
+        stats.getRemaining().get(),
+        stats.getCapacity().get()
+    );
+  }
+
+  private Map<String, Long> calculateNamespaceMetrics() {
+    Map<String, Long> metrics = new HashMap<>();
+    long totalPendingAtOmSide = calculatePendingSizes();
+    long totalOpenKeySize = calculateOpenKeySizes();
+    long totalCommittedSize = calculateCommittedSize();
+    long totalUsedNamespace = totalPendingAtOmSide + totalOpenKeySize + totalCommittedSize;
+    metrics.put("totalPendingAtOmSide", totalPendingAtOmSide);
+    metrics.put("totalOpenKeySize", totalOpenKeySize);
+    metrics.put("totalCommittedSize", totalCommittedSize);
+    metrics.put("totalUsedNamespace", totalUsedNamespace);
+    return metrics;
+  }
+
+  private StorageCapacityDistributionResponse buildStorageDistributionResponse(
+      List<DatanodeStorageReport> nodeStorageReports,
+      GlobalStorageReport storageMetrics,
+      Map<String, Long> namespaceMetrics) {
+    DeletedBlocksTransactionSummary scmSummary;
+    try {
+      scmSummary = scmClient.getDeletedBlockSummary();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    long totalPendingAtDnSide = nodeStorageReports.stream().mapToLong(DatanodeStorageReport::getPendingDeletions).sum();
+
+    DeletionPendingBytesByStage deletionPendingBytesByStage =
+        createDeletionPendingBytesByStage(namespaceMetrics.get("totalPendingAtOmSide"),
```

Comment:
`namespaceMetrics.get()` can return null, leading to NPE in createDeletionPendingBytesByStage().

We can use something like this - 
```
createDeletionPendingBytesByStage(
    namespaceMetrics.getOrDefault("totalPendingAtOmSide", 0L),
    scmSummary.getTotalBlockReplicatedSize(),
    totalPendingAtDnSide);
```